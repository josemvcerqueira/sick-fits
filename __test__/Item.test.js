import ItemComponent from "../components/Item";
import formatMoney from "../lib/formatMoney";
import { shallow } from "enzyme";

const fakeItem = {
	id: "ABC",
	title: "Gucci Shades",
	price: 50000,
	description: "This item will make you super popular",
	image: "gucci.jpg",
	largeImage: "largegucci.jpg"
};

describe("<Item/>", () => {
	const wrapper = shallow(<ItemComponent item={fakeItem} />);

	it("renders the image properly", () => {
		const img = wrapper.find("img");
		expect(img.props().src).toBe(fakeItem.image);
		expect(img.props().alt).toBe(fakeItem.title);
	});

	it("renders the price tag properly", () => {
		const PriceTag = wrapper.find("PriceTag");
		const price = formatMoney(fakeItem.price);

		expect(PriceTag.children().text()).toEqual(price);
		expect(wrapper.find("Title a").text()).toEqual(fakeItem.title);
	});

	it("renders the title properly", () => {
		expect(wrapper.find("Title a").text()).toEqual(fakeItem.title);
	});

	it("renders out the buttons properly", () => {
		const buttonList = wrapper.find(".buttonList");

		expect(buttonList.children()).toHaveLength(3);
		expect(buttonList.find("Link").exists()).toBe(true);
		expect(buttonList.find("AddToCart").exists()).toBe(true);
		expect(buttonList.find("DeleteItem").exists()).toBe(true);
	});
});
