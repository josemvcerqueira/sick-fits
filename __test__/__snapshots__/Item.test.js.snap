// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Gucci Shades"
    src="gucci.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC",
          },
        }
      }
    >
      <a>
        Gucci Shades
      </a>
    </Link>
  </Title>
  <PriceTag>
    $500
  </PriceTag>
  <p>
    This item will make you super popular
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC",
          },
        }
      }
    >
      <a>
        Edit ðŸ–‹
      </a>
    </Link>
    <AddToCart
      id="ABC"
    />
    <DeleteItem
      id="ABC"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
